### ISSUE TITLE
JavaCC 8.0.0: Table driven token manager dosn't handle ignore case 

### ISSUE THREAD

**Issue by mbrigl** [javacc/javacc#65]
The new table driven token manager ignore the definition IGNORE_CASE on tokens.

E.g. following definition will generate a parser that can parse _null_ but not _NULL_

`// Constants Tokens
TOKEN [ IGNORE_CASE ] :
{
  < NULL_TOKEN : "null" >
}
`

**Comment by new-javacc** [comment]
Did you test it with the interpreter?

**Comment by mbrigl** [comment]
Yes, I tried it. 

Parsing `null`
```
*** Starting in lexical state: DEFAULT
Token: NULL_TOKEN; image: "null" at: -1:1
```

Parsing `NULL`

```
Encountered token error at char: N
*** Starting in lexical state: DEFAULT
```

If you look at the _RStringLiteral.UpdateStringLiteralData_, it cannot work there is no information about case insensitive tokens to build the NFA. It handles only case insensitive globally if the option IGNORE_CASE is set

**Comment by new-javacc** [comment]
Looking

**Comment by new-javacc** [comment]
It works with -NO_DFA option. So the issue is when we generate the DFA for a string literal, we are not handing the IGNORE_CASE on token properly. Will fix it

**Comment by new-javacc** [comment]
OK A solution is to add another boolean array to the TokenizerData class and record any string literals that are token level ignore_case and then we can handle it in the driver to lower case the input.

**Comment by mbrigl** [comment]
Resolved with Pull Request #129
